File: LeetCode/univalued-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isUnivalTree(TreeNode root) {
        return isUnivalTree(root,root.val);
    }
    
    public boolean isUnivalTree(TreeNode root, int val){
        if(root == null){
            return true;
        }else{
            if(root.val == val){
                return (true && (isUnivalTree(root.left, val) && isUnivalTree(root.right,val)));
            }else{
                return false;
            }
        }
    }
}

File: LeetCode/univalued_binary_tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def isUnivalTree(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if (root != None):
            if (root.left != None and root.left.val != root.val):
                return False
            if (root.right != None and root.right.val != root.val):
                return False
            return (self.isUnivalTree(root.left) and self.isUnivalTree(root.right))
        else:
            return True
